docs/design/operator.md

```diff
@@ -0,0 +1,407 @@
+# Ceph CSI Operator Design Document
+
+## Introduction
+
+This document outlines the design of the Ceph-CSI Operator, which aims to
+provide Kubernetes native management interfaces for Ceph-CSI drivers (CephFS,
+RBD, and NFS) for Kubernetes based environments. The operator automates the
+deployment, configuration, and management of these drivers using new Kubernetes
+APIs defined as a set of Custom Resource Definitions (CRDs).
+
+## System Overview
+
+The Ceph-CSI Operator is a Kubernetes operator that offers a Kubernetes native
+management interfaces for Ceph CSI drivers. It provides a set of new Kubernetes
+APIs (CRDs) for governing deployment and management of Ceph-CSI drivers, as
+well as connection details to Ceph clusters. The operator ensures that the
+desired state specified in the CRDs is maintained in the cluster.
+
+## Assumptions and Dependencies
+
+Kubernetes cluster is available and operational. Ceph cluster is set up and
+accessible from the Kubernetes cluster. Users are familiar with Kubernetes and
+Ceph concepts. CRD and operator versions are compatible with the Kubernetes
+version in use.
+
+## General Constraints
+
+Resource limitations of the Kubernetes cluster (CPU, memory). Network latency
```

Comment:
How are these constraints specific to the Ceph-CSI operator? They seem applicable to anything in K8s. 

I'd suggest to use a bulleted list so each separate constraint is clear, and what the design point is for that constraint.